### DISCUSSION TITLE
Best Practices for Error Handling

### DISCUSSION THREAD

**Discussion by hhuseyinpay** [mobxjs/mobx.dart#986] [Q&A]
I've noticed that error handling in Flutter seems to be frequently overlooked. In particular, official examples lack try-catch statements. After conducting [research](https://github.com/search?q=mobx+language%3ADart&type=repositories&l=Dart), I couldn't find a repository that demonstrates effective error handling practices. Coming from a backend development background, where error handling is paramount, this situation surprises me. For instance, the Flutter HTTP library can throw a multitude of errors, yet it appears that exceptions are seldom handled properly. Is there a reason for this approach that I'm unaware of? Am I missing something in my understanding of Flutter's error management philosophy?

**Comment by fredgrott** [comment]
the repo data state mobx has examples and its a package itself.  Although it has not been updated in two years, one can still soft fork it and re-use.
